incubator/hnc/pkg/controllers/setup.go

```diff
@@ -3,30 +3,22 @@ package controllers
 import (
 	"fmt"
 
+	"github.com/kubernetes-sigs/multi-tenancy/incubator/hnc/pkg/config"
+	"k8s.io/apimachinery/pkg/runtime/schema"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/event"
 
-	"github.com/kubernetes-sigs/multi-tenancy/incubator/hnc/pkg/config"
 	"github.com/kubernetes-sigs/multi-tenancy/incubator/hnc/pkg/forest"
 )
 
 // Create creates all reconcilers.
 //
 // This function is called both from main.go as well as from the integ tests.
-func Create(mgr ctrl.Manager, f *forest.Forest, maxReconciles int) error {
+func Create(mgr ctrl.Manager, f *forest.Forest, maxReconciles int, newObjectController bool) error {
 	// Create all object reconcillers
-	objReconcilers := []NamespaceSyncer{}
-	for _, gvk := range config.GVKs {
-		or := &ObjectReconciler{
-			Client: mgr.GetClient(),
-			Log:    ctrl.Log.WithName("controllers").WithName(gvk.Kind),
-			Forest: f,
-			GVK:    gvk,
-		}
-		if err := or.SetupWithManager(mgr); err != nil {
-			return fmt.Errorf("cannot create %v controller: %s", gvk, err.Error())
-		}
-		objReconcilers = append(objReconcilers, or)
+	objReconcilers, err := createObjectReconcilers(newObjectController, mgr, f, config.GVKs)
```

Comment:
This isn't what I meant - the loop is exactly the same in both cases so that should be in the top level function. It's only creating an _individual_ reconciler that's different so that's the part should be in the new function, or at least that's what I though - do you disagree?

Reply:
Right, agreed!